src/neo/SmartContract/InteropService.Storage.cs

```diff
@@ -24,9 +24,34 @@ public static class Storage
             public static readonly InteropDescriptor PutEx = Register("System.Storage.PutEx", Storage_PutEx, GetStoragePrice, TriggerType.Application, CallFlags.AllowModifyStates);
             public static readonly InteropDescriptor Delete = Register("System.Storage.Delete", Storage_Delete, 0_01000000, TriggerType.Application, CallFlags.AllowModifyStates);
 
-            private static long GetStoragePrice(EvaluationStack stack)
+            private static long GetStoragePrice(ApplicationEngine engine)
             {
-                return (stack.Peek(1).GetByteLength() + stack.Peek(2).GetByteLength()) * GasPerByte;
+                var stack = engine.CurrentContext.EvaluationStack;
+                var key = stack.Peek(1);
+                var value = stack.Peek(2);
+                var newDataSize = value.IsNull ? 0 : value.GetByteLength();
+                if (!(engine.CurrentContext.EvaluationStack.Peek() is InteropInterface _interface))
+                    throw new InvalidOperationException();
+
+                StorageContext context = _interface.GetInterface<StorageContext>();
+                StorageKey skey = new StorageKey
+                {
+                    Id = context.Id,
+                    Key = key.GetSpan().ToArray()
+                };
+                var skeyValue = engine.Snapshot.Storages.TryGet(skey);
+                if (skeyValue is null || skeyValue.Value is null || skeyValue.Value.Length == 0)
+                    return (key.GetByteLength() + newDataSize) * GasPerByte;
+
+                var currentOccupiedBytes = skeyValue.Value.Length;
+                if (newDataSize <= currentOccupiedBytes)
+                {
+                    return 1 * GasPerByte;
```

Comment:
What is this `1`?

Reply:
Add 1 byte basic fee for the following case.
```
PUT(key, value111111111111111111111)
PUT(key, value1)   <----- fee = 1 byte * GasPerByte
```